### DISCUSSION TITLE
Using within Visual Studio

### DISCUSSION THREAD

**Discussion by SamChong015** [ggml-org/whisper.cpp#1146] [Q&A] [answered]
Hello, I am trying to use the header files within a C++ program I have made. I have the files downloaded (and it works in the command prompt) but I can't seem to figure out how to use the header files within my code.

**Comment by RndyP** [comment]
Add ggml.h, ggml.c, Whiper.h and Whisper.cpp to your project. Build. What happens?

**Reply by SamChong015** [reply]
When I build it the build fails and I get a bunch of errors from both ggml files.
![image](https://github.com/ggerganov/whisper.cpp/assets/119349070/c6a5cee3-8464-4746-8518-745fd38e877c)

**Reply by alonfaraj** [reply]
@SamChong015 can you share your project? On github maybe?  
It would be easier to figure out what's wrong..

**Reply by SamChong015** [reply]
I'll preface this with I don't have much experience with programming using files from other people. I made a GitHub Repo for my project (can be seen here https://github.com/SamChong015/JARVIS), but the ggml and whisper files do not show on the repository. I tried both with them held externally and adding it to the solution and with copying it directly to the source file or the project (both ways give me the same error with the ggml files).

**Reply by SamChong015** [reply]
I manged to get the files into the repo so you can see it, but I don't know how helpful it will be.

**Reply by SamChong015** [reply, marked as answer]
So I figured it out. Turns out I forgot to install cmake and now everything seems to be working fine. Thank you for the help!
